csrc/root_domain_map.cpp

```diff
@@ -102,12 +102,6 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     const TensorDomain* consumer,
     const std::unordered_set<IterDomain*>& root_dims_to_map,
     bool producer_to_consumer) const {
-  // Sanity check that the given producer and consumer domains are
-  // really the TensorDomains of the producer and consumer TensorViews
-  // given to the constructor.
-  TORCH_INTERNAL_ASSERT(producer_tv_->domain() == producer);
-  TORCH_INTERNAL_ASSERT(consumer_tv_->domain() == consumer);
-
```

Comment:
I don't think this check makes sense. If it must be the tensor domain of the given tensor view, why bother passing it as a parameter? In fact, it can be a replay of the tensor, so it is not the same tensor domain.

Reply:
We should clean this up, but the tensor view must be the same.

I think this was originally started with `ComputeAtRootDomainmap`, which doesn't need to have TensorViews to be passed in when mapping a producer TensorDomain and a consumer TensorDomain as all ID mappings are already created for the whole fusion during the constructor. 

Then, we also added `PairwiseRootDomainMap`, which is just a local pairwise mapping, as another implementation class of `RootDomainMap`. To do the pairwise mapping, we need to know the expression between the producer and consumer TensorDomains, but to grab the expression, we also need to have their respective TensorViews, so that's why `PariwiseRootDomainMap` needs to have them as constructor parameters. The assertions here are just sanity check the TensorDomains passed to `map` are indeed the ones associated with the `TensorViews` given at the constructor call.

If I were to rewrite this, I'd first change the `map` function to take producer and consumer TensorViews rather than TensorDomains.

Do these assertion cause any problem? I have other things that I wanted to clean up in this class, so I'd do it now if it's a blocker.

Reply:
Yes, it is a blocker, and unfortunately
> If I were to rewrite this, I'd first change the map function to take producer and consumer TensorViews rather than TensorDomains.
this is in the opposite direction as what I wanted. What I wanted is to pass in two tensor domains, and the tensor domain might not be the current TV's tensor domain, but a replay of it.

Reply:
Ah, OK, then let's remove the assertions.